src/Uno.UI.Composition/Composition/SkiaCompositionSurface.skia.cs

```diff
@@ -54,13 +66,58 @@ internal SkiaCompositionSurface(SKImage image)
 			{
 				try
 				{
-					_image = SKImage.FromEncodedData(stream);
-					return _image is null
-						? (false, "Failed to decode image")
-						: (true, "Success");
+					using var codec = SKCodec.Create(stream);
+					_infos = codec.FrameInfo;
+					var info = codec.Info;
+					info = new SKImageInfo(info.Width, info.Height, SKColorType.Bgra8888, SKAlphaType.Premul);
+					var bitmap = new SKBitmap(info);
+					_images = GC.AllocateUninitializedArray<SKImage>(_infos.Length);
+					var totalDuration = 0;
+					for (int i = 0; i < _infos.Length; i++)
+					{
+						var options = new SKCodecOptions(i);
+						codec.GetPixels(info, bitmap.GetPixels(), options);
+						var currentBitmap = SKImage.FromBitmap(bitmap);
+						if (currentBitmap is null)
+						{
+							_images = null;
+							return (false, "Failed to decode image");
+						}
+
+						_images[i] = currentBitmap;
+						totalDuration += _infos[i].Duration;
+					}
+
+					_totalDuration = totalDuration;
+
+					_ = Task.Run(() =>
+					{
+						// The created thread will probably live forever.
+						// Figure out when the SkiaCompositionSurface is no longer needed/used to terminate this thread.
+						CurrentFrame = 0;
+						while (_images is not null && _images.Length > 1)
```

Comment:
We don't support auto-repeat GIF here?

Note: currently this object will leak until the all frames have been consumed, so implementing support of auto-repeat would leak the object forever.

Reply:
I think leak concerns should be fixed now